toolchain/parse/handle_pattern_list.cpp

```diff
@@ -16,17 +16,22 @@ static auto HandlePatternListElement(Context& context, State pattern_state,
   context.PushStateForPattern(pattern_state, state.in_var_pattern);
 }
 
-auto HandlePatternListElementAsImplicit(Context& context) -> void {
-  HandlePatternListElement(context, State::BindingPattern,
-                           State::PatternListElementFinishAsImplicit);
-}
-
 auto HandlePatternListElementAsTuple(Context& context) -> void {
-  HandlePatternListElement(context, State::BindingPattern,
+  HandlePatternListElement(context, State::Pattern,
                            State::PatternListElementFinishAsTuple);
 }
 
-// Handles PatternListElementFinishAs(Implicit|Tuple).
+auto HandlePatternListElementAsExplicit(Context& context) -> void {
+  HandlePatternListElement(context, State::Pattern,
+                           State::PatternListElementFinishAsExplicit);
+}
+
+auto HandlePatternListElementAsImplicit(Context& context) -> void {
+  HandlePatternListElement(context, State::Pattern,
+                           State::PatternListElementFinishAsImplicit);
+}
+
+// Handles PatternListElementFinishAs(Tuple|Explicit|Implicit).
```

Comment:
Nit: would be nice for the order to match the above (line 10), and function order. I'm always a fan of lexical orderings, but mainly asking for consistency.